go/otgrpc/shared.go

```diff
@@ -18,6 +20,7 @@ type metadataReaderWriter struct {
 }
 
 func (w metadataReaderWriter) Set(key, val string) {
+	key = strings.ToLower(key)
```

Comment:
Looks like opentracing-go implementation is case sensitive. I'll open a PR.

https://github.com/opentracing/opentracing-go/blob/master/propagation.go#L160

```
func (c HTTPHeadersCarrier) Set(key, val string) {
	h := http.Header(c)
	h.Add(key, val)
}
```